NEWS.md

```diff
@@ -65,6 +65,9 @@ unit = "s")
 # data.table::fcase   1.49  1.55  1.67     1.71  1.73   1.86     5
 ```
 
+7. `.SDcols=is.numeric` now works; i.e., `SDcols=` accepts a function which is used to select the columns of `.SD`, [#3950](https://github.com/Rdatatable/data.table/issues/3950). Any function (even _ad hoc_) that returns scalar `TRUE`/`FALSE` for each column will do; e.g., `.SDcols=!is.character` will return _non_-character columns (_a la_ `Negate()`). Note that `patterns=` can still be used for filtering based on the column names.
```

Comment:
`patterns=` looks a bit misleading to me, looks like an argument of `[`, e.g. `dt[i, lapply(.SD, mean), patterns="^x"]`.

Reply:
Yep, thanks. I filed to #4642 to fix that.